agent/command/assume_ec2_role.go

```diff
@@ -0,0 +1,120 @@
+package command
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/evergreen-ci/evergreen/agent/internal"
+	"github.com/evergreen-ci/evergreen/agent/internal/client"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/evergreen-ci/utility"
+	"github.com/mitchellh/mapstructure"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+type ec2AssumeRole struct {
+	// The Amazon Resource Name (ARN) of the role to assume.
+	// Required.
+	RoleARN string `mapstructure:"role_arn" plugin:"expand"`
+
+	// A unique identifier that might be required when you assume a role in another account.
+	// This variable will only be accessible by admins and mainline.
+	ExternalId string `mapstructure:"external_id" plugin:"expand"`
+
+	// An IAM policy in JSON format that you want to use as an inline session policy.
+	Policy string `mapstructure:"policy" plugin:"expand"`
+
+	// The duration, in seconds, of the role session.
+	// Defaults to 900s (15 minutes).
+	DurationSeconds int `mapstructure:"duration_seconds"`
+
+	base
+}
+
+func ec2AssumeRoleFactory() Command   { return &ec2AssumeRole{} }
+func (r *ec2AssumeRole) Name() string { return "ec2.assume_role" }
+
+func (r *ec2AssumeRole) ParseParams(params map[string]interface{}) error {
+	if err := mapstructure.Decode(params, r); err != nil {
+		return errors.Wrapf(err, "error parsing '%s' params", r.Name())
+	}
+
+	return r.validate()
+}
+
+func (r *ec2AssumeRole) validate() error {
+	catcher := grip.NewSimpleCatcher()
```

Comment:
It seems like we can just return the error right away, I don't know if we need this catcher.